src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/Internal/HostTests.cs

```diff
@@ -628,6 +628,57 @@ public async Task HostStopAsyncUsesDefaultTimeoutIfGivenTokenDoesNotFire()
             }
         }
 
+        [Fact]
+        public async Task HostDoesNotStartHostedServiceIfCancellationIsRequested()
```

Comment:
Add 2 tests
1. One that verifies no hosted services start if the host calls `IHostApplicationLifetime.StopApplication` before calling StartAsync.
2. Another that uses IHostApplicationLifetime.StopApplication inside of an IHostedService.StartAsync (that's the scenario you were testing right?)

Reply:
Thank you for this! 

1. If I understand you correctly and read the code right, you are asking for a test [for this existing line of code](https://github.com/dotnet/runtime/blob/025e7f60cb355f9c44a5ff20784118e785891a4c/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L59) with `_applicationLifetime.ApplicationStopping` being cancelled? Or are you thinking in the lines of calling `IHostApplicationLifetime.StopApplication` as the hosted services [are being resolved from DI](https://github.com/dotnet/runtime/blob/025e7f60cb355f9c44a5ff20784118e785891a4c/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L60)?
2. I believe that this scenario is covered in `HostStopsGracefullyIfHostedServiceStopsTheApplication`, where the first hosted service [calls StopApplication](https://github.com/dotnet/runtime/pull/52517/files#diff-0004f2f896ea10dbe221c137338cf0bf30e116223a38d91911966c58b4d5b5beR673). Let me know if there is something I missed here!